src/overlays/actors/ovl_En_Ge3/z_en_ge3.c

```diff
@@ -28,49 +46,386 @@ const ActorInit En_Ge3_InitVars = {
+                     morphFrames);
+    this->animIndex = index;
+}
+
+void EnGe3_LookAtPlayer(EnGe3* this, PlayState* play) {
+    s16 yawDiff = this->picto.actor.yawTowardsPlayer - this->picto.actor.shape.rot.y; // temp not required
+
+    if ((ABS_ALT(yawDiff) <= 0x2300) && (this->picto.actor.xzDistToPlayer < 100.0f)) {
+        Actor_TrackPlayer(play, &this->picto.actor, &this->headRot, &this->torsoRot, this->picto.actor.focus.pos);
+    } else {
+        Math_SmoothStepToS(&this->headRot.x, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->headRot.y, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->torsoRot.x, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->torsoRot.y, 0, 6, 0x1838, 0x64);
+    }
+}
+
+void EnGe3_SetupPath(EnGe3* this, PlayState* play) {
+    this->curPointIndex = 0;
+
+    if (GERUDO_AVEIL_GET_PATH(&this->picto.actor) != GERUDO_AVEIL_PATH_NONE) {
+        this->path = &play->setupPathList[GERUDO_AVEIL_GET_PATH(&this->picto.actor)];
+    } else {
+        this->path = NULL;
+    }
+}
+
+/**
+ * @return true if no path or reached end of path
+ */
+s32 EnGe3_FollowPath(EnGe3* this) {
+    Path* path;
+    Vec3s* curPoint;
+    f32 diffX;
+    f32 diffZ;
+
+    path = this->path;
+
+    if (path == NULL) {
+        return true;
+    }
+
+    curPoint = Lib_SegmentedToVirtual(path->points);
+    curPoint += this->curPointIndex;
+    diffX = curPoint->x - this->picto.actor.world.pos.x;
+    diffZ = curPoint->z - this->picto.actor.world.pos.z;
+    this->picto.actor.world.rot.y = Math_Atan2S(diffX, diffZ);
+    Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.world.rot.y, 2, 0x7D0, 0xC8);
+
+    if ((SQ(diffX) + SQ(diffZ)) < SQ(10.0f)) {
+        this->curPointIndex++;
+        if (this->curPointIndex >= path->count) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void EnGe3_Scream(EnGe3* this) {
+    if ((s32)Rand_ZeroFloat(2.0f) == 0) {
+        Actor_PlaySfxAtPos(&this->picto.actor, NA_SE_VO_FPVO00);
+    } else {
+        Actor_PlaySfxAtPos(&this->picto.actor, NA_SE_VO_FPVO01);
+    }
+}
+
+void EnGe3_Idle(EnGe3* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    EnGe3_LookAtPlayer(this, play);
+
+    if (EnGe3_FollowPath(this)) {
+        EnGe3_SetupPath(this, play);
+    }
+}
+
+void EnGe3_ThrowPlayerOut(EnGe3* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.yawTowardsPlayer, 2, 0x400, 0x100);
+
+    if (this->actionTimer > 0) {
+        this->actionTimer--;
+    } else {
+        if (play->nextEntrance != play->setupExitList[GERUDO_AVEIL_GET_EXIT(&this->picto.actor)]) {
+            play->nextEntrance = play->setupExitList[GERUDO_AVEIL_GET_EXIT(&this->picto.actor)];
+            play->transitionTrigger = TRANS_TRIGGER_START;
+            play->transitionType = TRANS_TYPE_38;
+            gSaveContext.save.weekEventReg[80] &= (u8)~8;
```

Comment:
These generally don't match without the cast, there's a couple of hundred in the repo like this already. (It loads 2 bytes into the register instead of 1.)

Reply:
Oh interesting, havent seen that myself but understandable.